src/v/kafka/server/handlers/handlers.h

```diff
@@ -107,4 +110,92 @@ static constexpr size_t max_api_key(type_list<RequestTypes...>) {
     /// std::initializer_list
     return std::max({RequestTypes::api::key()...});
 }
+
+template<KafkaApi kk, security::audit::event_type vv>
+struct event_type_entry {
+    using k = kk;
+    static constexpr security::audit::event_type v = vv;
+};
+
+template<typename... T>
+struct event_type_map;
+
+template<KafkaApi kk, security::audit::event_type vv, typename... rest>
+struct event_type_map<event_type_entry<kk, vv>, rest...> {
+    template<KafkaApi k>
+    struct get {
+        static constexpr security::audit::event_type val = []() {
+            if constexpr (std::is_same_v<k, kk>) {
+                return vv;
+            } else {
+                return event_type_map<rest...>::template get<k>::val;
+            }
+        }();
+    };
+};
+
+using event_type_mapper = event_type_map<
```

Comment:
Wondering if this complexity is really necessary. The call sites can just pass the event type so why is there a map to look it up?

Reply:
to that point you could just then make methods like `audit_management_type(...)` one for each type and you'd have the same clarity, ensuring the callers call the right method.

Reply:
> to that point you could just then make methods like audit_management_type(...) one for each type and you'd have the same clarity, ensuring the callers call the right method.

This is kind of doing that though.  You call `get_audit_event_type<T>` and it provides the event type at compile time.

Can you expand maybe a little bit on what you mean?